sdk/nodejs/policy/proxy.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export function proxy<T>(toProxy: any): any {
+    return proxyHelper(toProxy, []);
+}
+
+function proxyHelper<T>(toProxy: any, props: (keyof T)[]): any {
```

Comment:
i'm pretty sure i'm understanding it.  but it prob wouldn't hurt to document this.  i.e. this array represents the sequence of property names accessed off the object (at least that's how i understand it).